internal/storage/image.go

```diff
@@ -781,57 +770,28 @@ func (svc *imageService) ResolveNames(systemContext *types.SystemContext, imageN
 		return nil, err
 	}
 
-	domain, remainder := splitDockerDomain(imageName)
-	if domain != "" {
-		// this means the image is already fully qualified
-		registry, err := sysregistriesv2.FindRegistry(systemContext, imageName)
-		if err != nil {
-			return nil, err
-		}
-		if registry != nil && registry.Blocked {
```

Comment:
Note that this looks up based on the user-specified name, but c/image/docker.newDockerClient` works with individual mirrors, so that’s a behavior change.

Reply:
Of course making CRI-O follow the documented semantics instead of its own is nice — but it’s a behavior change otherwise unrelated to this PR.

Reply:
Do you want me to drop it?

Reply:
I really don't like that CRI-O is occasionally working around containers/image.  `containers-registries.conf(5)` should be the source of truth.  On the other hand, I understand that behavioural changes may not be desired.

Reply:
(To be fair, historically c/image has only added these features _after_ they have been implemented/copy+pasted in Libpod/… .)

I’m leaning towards dropping this, but I’ll defer to CRI-O maintainers.

Reply:
To be more precise, by “leaning towards dropping this” I mean “dropping this change from the PR, i.e. leaving the `.Blocked` check in”. But it’s a very weak preference and up to CRI-O maintainers to decide.

Reply:
Thanks, @mtrmac!

@haircommander @mrunalp @rhatdan what do you think?  As Mirek mentioned, if a registry itself is blocked but has mirrors configured, CRI-O would not attempt to pull from these mirrors.

Reply:
I would be fine dropping this field as well TBH, instead of trying to work around it

Reply:
I see now that we have dropped it. sorry this review is all over the place :upside_down_face: